src/common/load-plugins.js

```diff
@@ -30,20 +29,10 @@ function loadPlugins(plugins, pluginSearchDirs) {
     plugin => typeof plugin === "string"
   );
 
-  const externalManualLoadPluginInfos = externalPluginNames.map(pluginName => {
-    let requirePath;
-    try {
-      // try local files
-      requirePath = resolve.sync(path.resolve(process.cwd(), pluginName));
-    } catch (e) {
-      // try node modules
-      requirePath = resolve.sync(pluginName, { basedir: process.cwd() });
-    }
-    return {
-      name: pluginName,
-      requirePath
-    };
-  });
+  const externalManualLoadPluginInfos = externalPluginNames.map(pluginName => ({
+    name: pluginName,
+    requirePath: eval("require").resolve(pluginName, { paths: [process.cwd()] })
+  }));
```

Comment:
This part might different with old one.

If  `pluginName = 'my-plugin'` not `pluginName = './my-plugin'`, result might different, will look for `my-plugin` package, insteadof `./my-plugin`, but no tests cover it.

Reply:
We should keep the existing behavior. I'd like to understand though what was the original motivation for using this module instead of `require.resolve`. Any ideas?

Reply:
https://github.com/prettier/prettier/pull/3536/files .

`require.resolve` supports `options` until `v8.9.0` https://nodejs.org/api/modules.html#modules_require_resolve_paths_request

I think that's the reason

Reply:
Logic of this part has restored.

Reply:
Can you please add a test for that case?

Reply:
How important is it to support the old behavior in the new release (#6888)?

Reply:
This should take a while, I don't know where pluginName come from yet.

Reply:
This line seems mean to test this, but did't catch it.

https://github.com/prettier/prettier/blob/master/tests_integration/__tests__/plugin-resolution.js#L148

Reply:
I add a print

```js
    console.log(
      JSON.stringify(
        {
          pluginName,
          "with-path.resolve": eval("require").resolve(
            path.resolve(process.cwd(), pluginName)
          ),
          "with-require.resolve(..., options)": eval("require").resolve(
            pluginName,
            {
              paths: [process.cwd()]
            }
          )
        },
        null,
        2
      )
    );
```

Got same result on `./automatic/node_modules/prettier-plugin-bar` and `automatic/node_modules/prettier-plugin-bar`, not sure where `.js` lost

```json
     {
       "pluginName": "./automatic/node_modules/prettier-plugin-bar",
       "with-path.resolve": "<repo>\\prettier\\tests_integration\\plugins\\automatic\\node_modules\\prettier-plugin-bar\\index.js",
       "with-require.resolve(..., options)": "<repo>\\prettier\\tests_integration\\plugins\\automatic\\node_modules\\prettier-plugin-bar\\index.js"
     }
```


```json
    {
       "pluginName": "automatic/node_modules/prettier-plugin-bar",
       "with-path.resolve": "<repo>\\prettier\\tests_integration\\plugins\\automatic\\node_modules\\prettier-plugin-bar\\index.js",
       "with-require.resolve(..., options)": "<repo>\\prettier\\tests_integration\\plugins\\automatic\\node_modules\\prettier-plugin-bar\\index.js"
     }
```

But I ran on cmd, this failed as expected

```text
<repo>\prettier\tests_integration\plugins>node
Welcome to Node.js v13.5.0.
Type ".help" for more information.
> require.resolve("automatic/node_modules/prettier-plugin-bar")
Uncaught Error: Cannot find module 'automatic/node_modules/prettier-plugin-bar'
Require stack:
- <repl>
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:17)
    at Function.resolve (internal/modules/cjs/helpers.js:78:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}
```

Reply:
@thorn0 have a look